### DISCUSSION TITLE
Is istio-cni not fully implemented yet? (unspecified.go)

### DISCUSSION THREAD

**Discussion by choigonyok** [istio/istio#48806] [Q&A] [answered]
I was studying Istio codebase because I wonder how istio-cni could redirect packets from/to sidecar proxy without linux capabilities.

But in `/cni/pkg/iptables/unspecified.go`, I knew that method `Program` is not implemented.

I think this method needs to set updated iptables to redirect from/to istio sidecar proxy.

Is istio-cni not fully implemented yet? or am i misunderstanding?

```
package plugin

import "errors"

// ErrNotImplemented is returned when a requested feature is not implemented.
var ErrNotImplemented = errors.New("not implemented")

// Program defines a method which programs iptables based on the parameters
// provided in Redirect.
func (ipt *iptables) Program(podName, netns string, rdrct *Redirect) error {
	return ErrNotImplemented
}
```

**Comment by howardjohn** [comment, marked as answer]
Thats only when you build on `!linux`. On linux it is implemented.

Istio-cni only runs on linux but its nice to build on OSX and windows for IDE support, etc
